jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java

```diff
@@ -1147,4 +1147,13 @@ public void close() throws Exception
     {
         stop();
     }
+
+    /**
+     * <p>Implementations of this interface are made aware of
+     * the {@code HttpClient} instance while it is starting.</p>
+     */
```

Comment:
Need more info than this.  Implementations of this interface that are: set as beans? added as listeners? discovered through ServiceLoader???

I think it should be on all beans (see below)

Reply:
Why? There is no need to do that.

Reply:
Because Aware is an interface on HttpClient. It looks like a general mechanism, not something just for some transport factories. If it is just for transport factories, then makes it a sub interface of transport factories.

You don't know that transport factories needed this until now, so why not just make it general and any bean added to the client can be aware.

Reply:
I have the opposite stance: I don't want to make it a generic mechanism when it is only needed in one case.
We can always make it generic, if we really need to, in the future.

It cannot be made part of `ClientConnectionFactory` because it's in `jetty-io`, while awareness is in `jetty-client`.

Reply:
In suggest you write the javadoc for this and you'll see that it is strange and difficult to document as a specific connection factory mechanism but with a generic class name. 

If you want to keep it like this, then it needs to be renamed to AwareClientConnectionFactory.

But it is simpler to name and explain AND LESS CODE, to just make it generic